aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -665,6 +665,80 @@ Tensor& NestedTensor_mul__Scalar(Tensor& self, const Scalar& other) {
   return NestedTensor_mul__Tensor(self, wrapped_scalar_tensor(other));
 }
 
+// Very rudimentary sum_dim for prototyping with torch_scatter.segment_reduce.
+Tensor NestedTensor_sum__dim(
+    const Tensor& self,
+    OptionalIntArrayRef opt_dims,
+    bool keepdim,
+    c10::optional<ScalarType> dtype) {
+  // Only allow reductions across the ragged (last) dim
+  auto dims = opt_dims.value_or(IntArrayRef{});
+  TORCH_CHECK(
+      dims.size() == 1,
+      "NestedTensor only allows reduction of a single dimension for now."
+  );
+  auto dim = maybe_wrap_dim(dims[0], self.dim());
+  TORCH_CHECK(
+      dim == self.dim() - 1,
+      "NestedTensor can only be reduced across the last dimension for now ",
+      "got dimension ",
+      dim,
+      " instead.");
+  // Always keep reduced dim for now
+  // This is to avoid the case where the nested tensors are 1D and keepdim=False
+  // making the nested tensors -> elements (e.g. sum(nt([1, 2 ,3], [4, 5]), -1) -> nt(6, 9))
+  TORCH_CHECK(keepdim, "NestedTensor always keeps the reduced dimension for now.");
+  // acc_dtype is not supported for now
+  TORCH_CHECK(!dtype, "NestedTensor does not support dtype argument for now.");
+
+  auto nt_input = get_nested_tensor_impl(self);
+  TORCH_CHECK(
+      nested_tensor_impl_is_contiguous(nt_input),
```

Comment:
side note: the definition of `nested_tensor_impl_is_contiguous()` is expanded to handle strides in Yifan's PR #80981 (which was reverted but will be relanded).

AFAICT this PR won't need updates so just FYI :)